qiskit-superstaq/qiskit_superstaq/superstaq_job.py

```diff
@@ -206,7 +214,8 @@ def _refresh_job(self) -> None:
             result = self._backend._provider._client.fetch_jobs(jobs_to_fetch)
             self._job_info.update(result)
 
-        self._update_status_queue_info()
+        if index is None:
+            self._update_status_queue_info()
```

Comment:
i think we want to do this for any index: e.g. maybe we're checking the last incomplete job in the batch, so if it's done we should update the overall status too
```suggestion
        self._update_status_queue_info()
```
however we'll have to update the logic in `_update_status_queue_info` to handle jobs we haven't retrieved yet (see next suggestion)